src/Products/PluginIndexes/DateRangeIndex/DateRangeIndex.py

```diff
@@ -238,12 +228,24 @@ def uniqueValues(self, name=None, withLengths=0):
                     else:
                         yield (key, len(value))
 
-    def _cache_key(self, catalog):
-        cid = catalog.getId()
-        counter = getattr(aq_base(catalog), 'getCounter', None)
-        if counter is not None:
-            return '%s_%s' % (cid, counter())
-        return cid
+    def _record_cache_key(self, record, resultset=None):
```

Comment:
I think I'd rename this here and on the UnIndex to something like `getRequestCacheKey` and signal that it is a bit more of a public API.

Than we could define one interface that specifies both getRequestCache and getRequestCacheKey as its only two methods and UnIndex implement it. That interface would really just be for documentation purposes, but I think in Zope we still like to have interfaces for everything ;)